src/commands/deploy/setPreDeployTaskForDotnet.ts

```diff
@@ -0,0 +1,75 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import { QuickPickItem, TextDocument, workspace, WorkspaceFolder } from 'vscode';
+import { ext } from 'vscode-azureappservice/out/src/extensionVariables';
+import * as constants from '../../constants';
+import { nonNullProp, nonNullValue } from '../../utils/nonNull';
+import * as workspaceUtil from '../../utils/workspace';
+import { getWorkspaceSetting, updateWorkspaceSetting } from '../../vsCodeConfig/settings';
+import * as tasks from '../../vsCodeConfig/tasks';
+import { IDeployWizardContext } from "../createWebApp/setAppWizardContextDefault";
+
+export async function setPreDeployTaskForDotnet(context: IDeployWizardContext): Promise<void> {
+    const fsPath: string = nonNullProp(context, 'fsPath');
+    const currentWorkspace: WorkspaceFolder | undefined = nonNullValue(workspaceUtil.getContainingWorkspace(fsPath));
+    // follow the publish output patterns, but leave out tfw
+    const dotnetOutputPath: string = path.join('bin', 'Debug', 'publish');
+
+    if (!getWorkspaceSetting('configurePreDeployTasks', currentWorkspace.uri.fsPath)) {
+        return;
+    }
+
+    const csProj = await workspaceUtil.findFilesByFileExtension(currentWorkspace.uri.fsPath, 'csproj');
+    if (csProj.length > 0) {
+        // if a publish task is already defined, then assume that we don't need this logic
+        if (getWorkspaceSetting(constants.configurationSettings.preDeployTask, currentWorkspace.uri.fsPath) !== 'publish') {
+            // if this doesn't have the publish preDeployTask, configure for .NET depoyment
+            const csProjDoc: TextDocument = await workspace.openTextDocument(csProj[0]);
+            const csProjJson: string = csProjDoc.getText();
+
+            const tfw: string = 'TargetFrameworks';
```

Comment:
I'm not sold on this logic. I think it would be confusing to prompt for a target framework out of the blue and this feels like an edge case.

I would rather handle the more standard case like this: https://github.com/Azure-Samples/dotnet-core-api/blob/master/TodoApi.csproj#L4 and wait for feedback/telemetry before we get into more complex cases.